samza-core/src/test/java/org/apache/samza/operators/TestJoinOperator.java

```diff
@@ -46,54 +47,47 @@
 import static org.mockito.Mockito.when;
 
 public class TestJoinOperator {
-  private final MessageCollector messageCollector = mock(MessageCollector.class);
   private final TaskCoordinator taskCoordinator = mock(TaskCoordinator.class);
-  private final Set<Integer> numbers = ImmutableSet.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
-
-  private StreamOperatorTask sot;
-  private List<Integer> output = new ArrayList<>();
   private final ApplicationRunner runner = mock(ApplicationRunner.class);
-
-
-  @Before
-  public void setup() throws Exception {
-    output.clear();
-
-    TaskContext taskContext = mock(TaskContext.class);
-    when(taskContext.getSystemStreamPartitions()).thenReturn(ImmutableSet
-        .of(new SystemStreamPartition("insystem", "instream", new Partition(0)),
-            new SystemStreamPartition("insystem2", "instream2", new Partition(0))));
-    Config config = mock(Config.class);
-
-    StreamApplication sgb = new TestStreamApplication();
-    sot = new StreamOperatorTask(sgb, runner);
-    sot.init(config, taskContext);
-  }
+  private final Set<Integer> numbers = ImmutableSet.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
 
   @Test
-  public void join() {
+  public void join() throws Exception {
+    StreamOperatorTask sot = createStreamOperatorTask();
+    List<Integer> output = new ArrayList<>();
+    MessageCollector messageCollector = envelope -> output.add((Integer) envelope.getMessage());
+
     // push messages to first stream
     numbers.forEach(n -> sot.process(new FirstStreamIME(n, n), messageCollector, taskCoordinator));
     // push messages to second stream with same keys
     numbers.forEach(n -> sot.process(new SecondStreamIME(n, n), messageCollector, taskCoordinator));
 
+
     int outputSum = output.stream().reduce(0, (s, m) -> s + m);
-    assertEquals(outputSum, 110);
+    assertEquals(110, outputSum);
```

Comment:
Same thing, I just had the order of expected and actual params reversed leading to confusing messages.